utils/fetchBridgeTokens.ts

```diff
@@ -0,0 +1,48 @@
+import { BridgeAction, BridgedEthereumToken, EthereumToken } from "@/types";
+import { AssetId, AssetInfoV41 as AssetInfo } from "@cennznet/types";
+import fetchEthereumTokens from "@/utils/fetchEthereumTokens";
+import { ETH_CHAIN_ID } from "@/constants";
+import { Api } from "@cennznet/api";
+import { hexToString } from "@polkadot/util";
+
+type BridgeTokens<T> = T extends "Deposit"
+	? EthereumToken
+	: T extends "Withdraw"
+	? BridgedEthereumToken
+	: never;
+
+export default async function fetchBridgeTokens<T extends BridgeAction>(
+	api: Api,
+	action: BridgeAction
+): Promise<BridgeTokens<T>[]> {
+	if (action === "Deposit")
+		return (await Promise.resolve(fetchEthereumTokens(ETH_CHAIN_ID))).sort(
+			(a, b) => (a.symbol > b.symbol ? 1 : -1)
+		) as BridgeTokens<T>[];
+
+	const registeredAssets: [AssetId, AssetInfo][] = await (
```

Comment:
can we store the registeredAsset in browser storage.. so that we don't need to wait to get registered asset on every reload/ call to this function..